NYPL-Simplified/Simplified-iOS
# SIMPLY-2617 Read non-DRM books using Readium 2

**What's this do?**
Opens up non-DRM'ed books in the SimplyE collection using R2. This is the first step toward integrating R2 into SimplyE and there are many things left to-do. It follows a similar architecture per the R2 test app, which for now makes the most sense, although that's subject to change. For example, the `ReaderFactory` class might not be needed or might need to be renamed, but at this time I don't know if that's going to be the case. Similarly, UI for TOC and Settings panels are also lifted entirely from the test app, but obviously that code duplication will be addressed in later tickets. Essentially this is just a first iteration that will necessarily receive further broad strokes.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SIMPLY-2617

**How should this be tested? / Do these changes have associated tests?**
To test this you can follow these steps:
- select the "SimplyE Collection" in the Catalog tab
- download a book and select Read
- the book should open and the UI will look slightly different: e.g. The footer UI does not show the chapter name; clicking on the Font panel icon on the top right will show a different UI for the font selection. All these issues have been filed and will be addressed in separate tickets.

**Dependencies for merging? Releasing to production?**
Merging to a feature branch. Before merging to develop we'll need to address all the tickets in the epic.

**Has the application documentation been updated for these changes?**
yes: https://github.com/NYPL-Simplified/Simplified/wiki/R2-Integration-on-iOS

**Did someone actually run this code to verify it works?**
@ettore
